torch/package/file_structure_representation.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+from typing import Dict, List
+from os import sep as os_separator
+
+from .glob_group import GlobPattern, GlobGroup
+
+
+class Directory:
+    """A file structure representation. Organized as Directory nodes that have lists of
```

Comment:
Might want to note that this is returned from ``:meth:`PackageExporter.file_structure` `` and ``:meth:`PackageImporter.file_structure` `` so that folks looking at this class context-free understand how it is used.